wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/text/TextStyles.kt

```diff
@@ -0,0 +1,490 @@
+ * Each argument should be within 0 and 1. If a number is outside the range, the function will automatically default it
+ * to the maximum or minimum value, depending on which is nearer to the provided value.
+ *
+ * @param red The red channel of the color
+ * @param green The green channel of the color
+ * @param blue The blue channel of the color
+ *
+ * @return A [Color] created from the provided RGB Channels
+ *
+ * @author NoComment1105
+ */
+@TextDslMarker
+@Suppress("MagicNumber")
+public fun Color(red: Double, green: Double, blue: Double): Color = Color(
+    (red.coerceIn(0.0, 1.0) * 255).roundToInt(),
+    (green.coerceIn(0.0, 1.0) * 255).roundToInt(),
+    (blue.coerceIn(0.0, 1.0) * 255).roundToInt(),
+)
+
+/**
+ * A color class that can transform RGB values into color codes.
+ *
+ * @property value The RGB value to convert to a color code.
+ *
+ * @author NoComment1105
+ */
+@TextDslMarker
+@JvmInline
+public value class Color(public val value: Int) {
+    public companion object {
+        /** Minecraft's native black color. */
+        public val BLACK: Color = Color(0x000000)
+        /** Minecraft's native dark blue color. */
+        public val DARK_BLUE: Color = Color(0x0000AA)
+        /** Minecraft's native dark green color. */
+        public val DARK_GREEN: Color = Color(0x00AA00)
+        /** Minecraft's native dark aqua color. */
+        public val DARK_AQUA: Color = Color(0x00AAAA)
+        /** Minecraft's native dark red color. */
+        public val DARK_RED: Color = Color(0xAA0000)
+        /** Minecraft's native dark purple color. */
+        public val DARK_PURPLE: Color = Color(0xAA00AA)
+        /** Minecraft's native gold color. */
+        public val GOLD: Color = Color(0xFFAA00)
+        /** Minecraft's native grey color. */
+        public val GREY: Color = Color(0xAAAAAA)
+        /** Minecraft's native dark grey color. */
+        public val DARK_GREY: Color = Color(0x555555)
+        /** Minecraft's native blue color. */
+        public val BLUE: Color = Color(0x5555FF)
+        /** Minecraft's native green color. */
+        public val GREEN: Color = Color(0x55FF55)
+        /** Minecraft's native aqua color. */
+        public val AQUA: Color = Color(0x55FFFF)
+        /** Minecraft's native red color. */
+        public val RED: Color = Color(0xFF5555)
+        /** Minecraft's native light purple color. */
+        public val LIGHT_PURPLE: Color = Color(0xFF55FF)
+        /** Minecraft's native yellow color. */
+        public val YELLOW: Color = Color(0xFFFF55)
+        /** Minecraft's native white color. */
+        public val WHITE: Color = Color(0xFFFFFF)
+
+        /**
+         * Gets a color from [TextColor] and converts it to [Color].
+         *
+         * @param color The [TextColor] to convert
+         * @return A [Color] from [color]
+         *
+         * @author NoComment1105
+         */
+        public fun fromTextColor(color: TextColor): Color {
```

Comment:
perhaps also a `fromMapColor` and `fromDyeColor`, although that last one has 3 collors associated, `signColor` might be the most appropriate (given that the Mojmap name is "getTextColor")?

It might also be best to then call them just `from` cause class names can be different in different mappings

Reply:
In favor of renaming all to `from`